### PR TITLE
Add support for using the extension exclusively for formatting

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

This PR enables users to specify `"ruff.lint.enable": false` to use the LSP (or extension) solely for formatting.

"Organize Imports" is still supported even when the linter is disabled.

Closes: https://github.com/astral-sh/ruff-vscode/issues/329.

Depends on: https://github.com/astral-sh/ruff-lsp/pull/361.

## Test Plan

- Ran the VS Code extension.
- Verified that diagnostics appeared for unused imports.
- Set `"ruff.lint.enable": false`.
- Verified that the diagnostics disappeared.
- Ran "Organize Imports".
- Verified that the imports were sorted.
- Ran "Fix All".
- Verified that the imports (which were unused) were not removed.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -56,3 +42,82 @@ module = [
   "pylsp_jsonrpc.*",
 ]
 ignore_missing_imports = true
+
+[tool.ruff]
```
Comment (dhruvmanila) [inline pyproject.toml:46]: Is this change relevant for this PR? There's also the `ruff.toml` file which probably contains the same settings as mentioned here.

**Review by dhruvmanila — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (charliermarsh) [reply pyproject.toml:46]: Thanks, these were stupidly checked-in from when I was doing some testing of our JSON Schema...

File: README.md [hunk 2]
```diff
@@ -67,17 +67,18 @@ This requires Ruff version `v0.1.3` or later.
...
+| interpreter                          | `[]`              | Path to a Python interpreter to use to run the linter server.                                                                                                                                                                                                     |
+| lint.args                            | `[]`              | Additional command-line arguments to pass to `ruff check`, e.g., `"args": ["--config=/path/to/pyproject.toml"]`. Supports a subset of Ruff's command-line arguments, ignoring those that are required to operate the LSP, like `--force-exclude` and `--verbose`. |
+| line.enable                          | `true`            | Whether to enable linting. Set to `false` to use Ruff exclusively as a formatter.                                                                                                                                                                                 |
```
Comment (MichaReiser) [inline README.md:78]: Shouldn't this be `lint.enable`?

Comment (charliermarsh) [reply README.md:78]: No! no no no!

**Pull request merged**